op-node/node/bvm_api_helper.go

```diff
@@ -85,31 +89,62 @@ func getBTCFinalityForBlockNum(ctx context.Context, blockNum uint64, driver driv
 
 	log.Trace("going to query for keystone", "keystone", spew.Sdump(l2Keystone), "nextKeystone", spew.Sdump(nextKeystone), "prevKeystoneHash", hex.EncodeToString(prevKeystoneHash[:]))
 
-	l2KeystonesToQuery := make([]hemi.L2Keystone, 1)
-	l2KeystonesToQuery[0] = *l2Keystone
+	client := &http.Client{}
+	kssHash := hemi.L2KeystoneAbbreviate(*l2Keystone).Hash()
+	u := fmt.Sprintf("%v/v%v/keystonefinality/%v",
+		bfgURL, bfgapi.APIVersion, kssHash)
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, u, nil)
+	if err != nil {
+		return emptyFin, err
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return emptyFin, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return emptyFin, fmt.Errorf("unexpected status code: %v", resp.StatusCode)
+	}
+
+	fin := bfgapi.L2KeystoneBitcoinFinalityResponse{}
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return emptyFin, err
+	}
+
+	if err = resp.Body.Close(); err != nil {
+		return emptyFin, err
+	}
+
+	if err = json.Unmarshal(body, &fin); err != nil {
+		return emptyFin, err
+	}
```

Comment:
This can be simplified by using `json.NewDecoder` - Additionally, this removes the need to buffer the entire body in memory before unmarshaling, and is overall far more performant.
```suggestion
	fin := bfgapi.L2KeystoneBitcoinFinalityResponse{}
	if err := json.NewDecoder(resp.Body).Decode(&fin); err != nil {
		return emptyFin, err
	}
```

Removed `resp.Body.Close()` as that will be handled in defer - we can keep it if error handling is required.

Reply:
looks good.  I don't think that we need to `Close()` and return that error; I think that the `defer` is fine.

Reply:
I love smaller code :)